'go test' fails since 2.14.3 in OpenBuildService

Hi,

trying to build gh client in [build.opensuse.org](https://build.opensuse.org/package/show/home:computersalat:devel:scm/gh).
But starting with version 2.14.3 the `go test` is failing. Maybe because of an Update to the `HTTPClient` test.

```
[   68s] go test ./...
[   72s] --- FAIL: TestNewHTTPClient (0.01s)
[   72s]     --- FAIL: TestNewHTTPClient/github.com_in_verbose_mode (0.00s)
[   72s]         http_client_test.go:203: 
[   72s]                Error Trace:    http_client_test.go:203
[   72s]                Error:          Not equal: 
[   72s]                                expected: "* Request at <time>\n* Request to http://<host>:<port>\n> GET / HTTP/1.1\n> Host: github.com\n> Accept: application/vnd.github.merge-info-preview+json, application/vnd.github.nebula-preview\n> Authorization: token ████████████████████\n> Content-Type: application/json; charset=utf-8\n> Time-Zone: <timezone>\n> User-Agent: GitHub CLI v1.2.3\n\n< HTTP/1.1 204 No Content\n< Date: <time>\n\n* Request took <duration>\n"
[   72s]                                actual  : "* Request at <time>\n* Request to http://<host>:<port>\n> GET / HTTP/1.1\n> Host: github.com\n> Accept: application/vnd.github.merge-info-preview+json, application/vnd.github.nebula-preview\n> Authorization: token ████████████████████\n> Content-Type: application/json; charset=utf-8\n> User-Agent: GitHub CLI v1.2.3\n\n< HTTP/1.1 204 No Content\n< Date: <time>\n\n* Request took <duration>\n"
[   72s]                            
[   72s]                                Diff:
[   72s]                                --- Expected
[   72s]                                +++ Actual
[   72s]                                @@ -7,3 +7,2 @@
[   72s]                                 > Content-Type: application/json; charset=utf-8
[   72s]                                -> Time-Zone: <timezone>
[   72s]                                 > User-Agent: GitHub CLI v1.2.3
[   72s]                Test:           TestNewHTTPClient/github.com_in_verbose_mode
[   72s]     --- FAIL: TestNewHTTPClient/github.com_in_verbose_mode#01 (0.00s)
[   72s]         http_client_test.go:203: 
[   72s]                Error Trace:    http_client_test.go:203
[   72s]                Error:          Not equal: 
[   72s]                                expected: "* Request at <time>\n* Request to http://<host>:<port>\n> GET / HTTP/1.1\n> Host: github.com\n> Accept: application/vnd.github.merge-info-preview+json, application/vnd.github.nebula-preview\n> Authorization: token ████████████████████\n> Content-Type: application/json; charset=utf-8\n> Time-Zone: <timezone>\n> User-Agent: GitHub CLI v1.2.3\n\n< HTTP/1.1 204 No Content\n< Date: <time>\n\n* Request took <duration>\n"
[   72s]                                actual  : "* Request at <time>\n* Request to http://<host>:<port>\n> GET / HTTP/1.1\n> Host: github.com\n> Accept: application/vnd.github.merge-info-preview+json, application/vnd.github.nebula-preview\n> Authorization: token ████████████████████\n> Content-Type: application/json; charset=utf-8\n> User-Agent: GitHub CLI v1.2.3\n\n< HTTP/1.1 204 No Content\n< Date: <time>\n\n* Request took <duration>\n"
[   72s]                            
[   72s]                                Diff:
[   72s]                                --- Expected
[   72s]                                +++ Actual
[   72s]                                @@ -7,3 +7,2 @@
[   72s]                                 > Content-Type: application/json; charset=utf-8
[   72s]                                -> Time-Zone: <timezone>
[   72s]                                 > User-Agent: GitHub CLI v1.2.3
[   72s]                Test:           TestNewHTTPClient/github.com_in_verbose_mode#01
[   72s] FAIL
[   72s] FAIL   github.com/cli/cli/v2/api       0.017s
```

IMHO the  build service does not allow to go online.
Is there a possibility to improve the test in a way that does not need to go online ?
Or are there any other ideas instead of `skipping` the tests.

Thank you

--- a/api/http_client_test.go
+++ b/api/http_client_test.go
@@ -182,6 +182,7 @@ func TestNewHTTPClient(t *testing.T) {
 			require.NoError(t, err)
 
 			req, err := http.NewRequest("GET", ts.URL, nil)
+			req.Header.Set("time-zone", "Europe/Amsterdam")
 			req.Host = tt.host
 			require.NoError(t, err)
 

